template/FormlessTemplate.purs

```diff
@@ -0,0 +1,148 @@
+module FormlessTemplate where
+
+import Prelude
+
+import DOM.HTML.Indexed.InputType (InputType(..))
+import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Effect.Aff.Class (class MonadAff)
+import Formless as F
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Template.DataAndValidation (NAME_FIELD, _name, minLength)
+import Type.Row (type (+))
+
+type FormFieldsRow f =
+  ( NAME_FIELD f
+  + ()
+  )
+
+type FormFields = { | FormFieldsRow F.OutputType }
+
+newtype Form r f = Form (r (FormFieldsRow f))
+derive instance newtypeForm' :: Newtype (Form r f) _
+
+-- Form component types
+
+type Input = Unit
+type AddedState = ( additionalState :: Maybe Int )
+data Action
+  = DoStuff
+  | Initialize
+  | Finalize
+  | Receive Input
+
+data Query a
+  = Command a
+  | Reply (Unit -> a)
+
+type Message = FormFields
+type ChildSlots =
+  ()
+type SelfSlot index = F.Slot Form Query ChildSlots Message index
+
+component
+  :: forall m
+   . MonadAff m
+  => F.Component Form Query ChildSlots Input FormFields m
+component = F.component mkInput $ F.defaultSpec
+  { render = render
+  , handleAction = handleAction
+  , handleEvent = handleEvent
+  , handleQuery = handleQuery
+  , initialize = Just Initialize
+  , finalize = Just Finalize
+  , receive = Just <<< Receive
+  }
+  where
+  -- Converts the Input value passed in by the parent component
+  -- into the Formless' Input value.
+  mkInput :: Input -> F.Input Form AddedState m
+  mkInput _ =
+    -- the two values here are for Formless
+    { validators: Form
+        { name: minLength 7
+        }
+    , initialInputs: Nothing -- when Nothing, will use `Initial` type class
+
+    -- everything else below comes from our `AddedState` rows:
+    , additionalState: Just 5
+    }
+
+  render
+    :: F.PublicState Form AddedState
+    -> F.ComponentHTML Form Action ChildSlots m
+  render st =
+    HH.div_
+      [ HH.p_ [ HH.text $ "Validity: " <> show st.validity ]
+      , HH.p_ [ HH.text $ "Dirty: " <> show st.dirty ]
+      , HH.p_ [ HH.text $ "Being Submitted: " <> show st.submitting ]
```

Comment:
Indicates whether the 'submit' button has been clicked and the form's content is still being validated one last time before submission is accepted.